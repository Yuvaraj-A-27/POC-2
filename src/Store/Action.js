import { ACTIVE_USER, ADD_TO_CART, CART_ACTIVE, CREATE_USER, CURRENT_PRODUCT, INITIAL_CATEGORY, INITIAL_PRODUCT, INITIAL_USER_DETAIL, LOGIN_ACTIVE, PRODUCT_POP_ACTIVE, PROFILE_ACTIVE, REGISTER_ACTIVE } from "./ActionType"

export const initialCategory = (value) =>{
    return{
        type: INITIAL_CATEGORY,
        payload : value
    }
}

export const initialProduct = (value) =>{
    return{
        type: INITIAL_PRODUCT,
        payload : value
    }
}

export const initialUserDetail = (value) =>{
    return{
        type:INITIAL_USER_DETAIL,
        payload : value
    }
}


export const loginActive = ()=>{
    return{
        type: LOGIN_ACTIVE,
    }
}

export const activeUser = (value)=>{
    return{
        type: ACTIVE_USER,
        payload: value
    }
}

export const productPopUpActive = ()=>{
    return{
        type: PRODUCT_POP_ACTIVE,
    }
}

export const currentProduct = (id) =>{
    return{
        type: CURRENT_PRODUCT,
        payload : id
    }
}

export const addToCart = (productId, userId) =>{
    return{
        type:ADD_TO_CART,
        payload: {
            productId,
            userId
        }
    }
}

export const profileActive = ()=>{
    return{
        type:PROFILE_ACTIVE,
    }
}

export const cartActive = () =>{
    return{
        type: CART_ACTIVE,
    }
}

export const registerActive = () =>{
    return{
        type : REGISTER_ACTIVE,
    }
}

export const createUser = (value) =>{
    return{
        type : CREATE_USER,
        payload : value
    }
}