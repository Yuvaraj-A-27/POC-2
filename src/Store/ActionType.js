export const INITIAL_CATEGORY = 'INITIAL_CATEGORY'
export const INITIAL_PRODUCT = 'INITIAL_PRODUCT'
export const INITIAL_USER_DETAIL = 'INITIAL_USER_DETAIL'
export const LOGIN_ACTIVE = 'LOGIN_ACTIVE'
export const ACTIVE_USER = 'ACTIVE_USER'
export const PRODUCT_POP_ACTIVE ="PRODUCT_POP_ACTIVE"
export const CURRENT_PRODUCT = 'CURRENT_PRODUCT'