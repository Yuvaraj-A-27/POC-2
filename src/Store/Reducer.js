import { ACTIVE_USER, ADD_TO_CART, CART_ACTIVE, CREATE_USER, CURRENT_PRODUCT, INITIAL_CATEGORY, INITIAL_PRODUCT, INITIAL_USER_DETAIL, LOGIN_ACTIVE, LOGIN_ERROR, LOGIN_ERROR_TYPING, PRODUCT_POP_ACTIVE, PROFILE_ACTIVE, REGISTER_ACTIVE } from "./ActionType";

const initialState ={
    category : [],
    product : [],
    userDetail : [],
    loginActive : false,
    productPopUpActive : false,
    profileActive : false,
    cartActive : false,
    registerActive : false,
    // loginError : false,
    activeUserDetail : '',
    currentProduct : 1,
    cart:[],
}

const Reducer = (state = initialState, action)=>{
    switch(action.type){
        case INITIAL_CATEGORY: return{
            ...state,
            category : [...action.payload]
        }

        case INITIAL_PRODUCT:return{
            ...state,
            product : [...action.payload]
        }

        case INITIAL_USER_DETAIL:return{
            ...state,
            userDetail : [...action.payload]
        }

        case LOGIN_ACTIVE: return{
            ...state,
            loginActive : !state.loginActive,
            productPopUpActive : false,
            profileActive:false,
            cartActive : false,
            registerActive : false,
        }

        case PRODUCT_POP_ACTIVE: return{
            ...state,
            productPopUpActive : !state.productPopUpActive,
            loginActive : false,
            profileActive:false,
            cartActive : false,
            registerActive : false,
        }

        case ACTIVE_USER: return{
            ...state,
            activeUserDetail : action.payload
        }

        case CURRENT_PRODUCT: return{
            ...state,
            currentProduct : action.payload
        }

        case ADD_TO_CART : return{
            ...state,
            cart : [...state.cart, action.payload]
        }

        case PROFILE_ACTIVE: return{
            ...state,
            profileActive : !state.profileActive,
            loginActive : false,
            productPopUpActive:false,
            cartActive : false,
            registerActive : false,
        }

        case CART_ACTIVE: return{
            ...state,
            cartActive : !state.cartActive,
            loginActive : false,
            productPopUpActive:false,
            profileActive : false,
            registerActive : false,
        }

        case REGISTER_ACTIVE : return{
            ...state,
            registerActive : !state.registerActive,
            cartActive : false,
            loginActive : false,
            productPopUpActive:false,
            profileActive : false
        }

        case CREATE_USER : return{
            ...state,
            userDetail : [...state.userDetail, action.payload]
        }

        // case LOGIN_ERROR : return{
        //     ...state,
        //     loginError : true
        // }

        // case LOGIN_ERROR_TYPING : return{
        //     ...state,
        //     loginError : false
        // }

        default : return state
    }
}

export default Reducer;