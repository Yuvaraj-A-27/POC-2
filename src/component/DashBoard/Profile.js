import { Dialog, DialogContent, DialogContentText, DialogTitle, Grid, makeStyles } from '@material-ui/core';
import React from 'react'
import { connect } from 'react-redux';
import { profileActive } from '../../Store/Action';

const useStyles = makeStyles(()=>({
    dialogPaper:{
        maxWidth: '800px',
        maxHeight: '1800px',
        minHeight: '500px',
    },
    content: {
        marginLeft:'30px',
        marginTop : '-20px'
    }
}))

function Profile(props){
    const classes = useStyles()

    // console.log(props.activeUserDetail);
    const detail = props.activeUserDetail? props.activeUserDetail[0]  : null
    if(detail){
    return(
        <Dialog
        open={props.profileActive}
        onClose={props.profileActiveHandler}
        fullWidth='true'
        maxWidth ='lg'
        classes={{ paperWidthLg: classes.dialogPaper }}
        >
            <DialogTitle>
                <h2>Profile - {detail.name.firstname} {detail.name.lastname}</h2>
            </DialogTitle>
            <DialogContent className={classes.content}>
                <DialogContentText>
                    <Grid container spacing={3}>
                        <Grid item xs={3}>
                            <strong data-testid ='userId' >User ID : </strong>
                        </Grid>
                        <Grid item xs={8}>
                            {detail.id}
                        </Grid>
                        <Grid item xs={3}>
                            <strong>Email : </strong>
                        </Grid>
                        <Grid item xs={8}>
                            {detail.email}
                        </Grid>
                        <Grid item xs={3}>
                            <strong>User Name : </strong>
                        </Grid>
                        <Grid item xs={8}>
                            {detail.username}
                        </Grid>
                        <Grid item xs={3}>
                            <strong>First Name : </strong>
                        </Grid><Grid item xs={8}>
                            {detail.name.firstname}
                        </Grid>
                        <Grid item xs={3}>
                            <strong>Last Name : </strong>
                        </Grid>
                        <Grid item xs={8}>
                            {detail.name.lastname}
                        </Grid>
                        <Grid item xs={3}>
                            <strong>Address : </strong>
                        </Grid>
                        <Grid item xs={8}>
                            {detail.address.number}, {detail.address.street}, {detail.address.city}, ZipCode-{detail.address.zipcode}
                        </Grid>
                        <Grid item xs={3}>
                            <strong>Phone : </strong>
                            </Grid>
                        <Grid item xs={8}>
                            {detail.phone}
                        </Grid>
                    </Grid>
                </DialogContentText>
            </DialogContent>
        </Dialog>
    );
    }
    return null   // to avoid error in testing  --- undefined and null error
}

const mapStateToProps = state =>{
    return{
        profileActive : state.profileActive,
        activeUserDetail : state.activeUserDetail
    }
}
const mapDispatchToProps = dispatch =>{
    return{
        profileActiveHandler : ()=>dispatch(profileActive())
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(Profile);
