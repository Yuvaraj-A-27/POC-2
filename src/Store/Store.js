import { createStore } from "redux"
import persistReducer from "redux-persist/es/persistReducer"
import persistStore from "redux-persist/es/persistStore"
import Reducer from "./Reducer"
import storage from 'redux-persist/lib/storage'


const persistConfig = {
    key:'root',
    storage,
}

const persistorReducer = persistReducer(persistConfig, Reducer)

export const Store = createStore(persistorReducer)
export const Persistor = persistStore(Store)