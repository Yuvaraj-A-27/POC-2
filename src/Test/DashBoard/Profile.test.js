import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Profile from '../../component/DashBoard/Profile'
import { PROFILE_ACTIVE } from '../../Store/ActionType'
import { profileActive } from '../../Store/Action'


const middleware  = {}
const mockStore = configureStore(middleware)
const initialState = {}
const Store = mockStore(initialState)



// const propsData = [{
//     address:{
//         city: "kilcoole",
//         geolocation: {lat: "-37.3159", long: "81.1496"},
//         number: 7682,
//         street: "new road",
//         zipcode: "12926-3874",
//     },
//     email: "john@gmail.com",
//     id: 1,
//     name:{
//         firstname: "john",
//         lastname: "doe",
//     },
//     password: "m38rmF$",
//     phone: "1-570-236-7033",
//     username: "johnd"
// }]

test('Profile component snapshot test', ()=>{
    const {} = render(
        <Provider store={Store}>
            <Profile />
        </Provider>
    )
    // setTimeout(() => {
    //     const profileActive = ()=>({type:PROFILE_ACTIVE})
    //     Store.dispatch(profileActive)
    // }, 5000);

    // const dispatchHandler = ()=>{
    //     Store.dispatch(profileActive)
    // }
    
    const profile = renderer.create(
        <Provider store={Store}>
            <Profile  />
        </Provider>
    )
    // profile.getInstance.dispatchHandler()
    expect(profile.toJSON()).toMatchSnapshot()
})