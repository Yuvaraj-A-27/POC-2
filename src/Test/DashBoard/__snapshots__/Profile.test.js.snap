// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component snapshot test 1`] = `null`;
